src/partr.c

```diff
@@ -405,6 +421,15 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
             int8_t gc_state = jl_gc_safe_enter(ptls);
             uv_mutex_lock(&ptls->sleep_lock);
             while (may_sleep(ptls)) {
+                if (ptls->tid == 0 && wait_empty) {
+                    task = wait_empty;
```

Comment:
Is it possible other threads are still running tasks, and we should keep waiting?

Reply:
I decided to leave handling threads correctly in the same way to a future PR, once this is approved and merged